checks/libs/win/winpdh.py

```diff
@@ -1,37 +1,63 @@
 import time
 import win32pdh
+import _winreg
 
-DATA_POINT_INTERVAL = 0.01
+DATA_POINT_INTERVAL = 0.10
 
 class WinPDHCounter(object):
-    def __init__(self, class_name, instance_name):
-        self._class_name = class_name
-        self._instance_name = instance_name
+    # store the dictionary of pdh counter names
+    pdh_counter_dict = {}
+
+    def __init__(self, class_name, instance_name, log):
+        self._get_counter_dictionary()
+        self._class_name = win32pdh.LookupPerfNameByIndex(None, int(WinPDHCounter.pdh_counter_dict[class_name]))
+        self._instance_name = win32pdh.LookupPerfNameByIndex(None, int(WinPDHCounter.pdh_counter_dict[instance_name]))
+
+        self._is_single_instance = False
         self.hq = win32pdh.OpenQuery()
+        self.logger = log
+        self.counterdict = {}
+        counters, instances = win32pdh.EnumObjectItems(None, None, self._class_name, win32pdh.PERF_DETAIL_WIZARD)
+        if len(instances) > 0:
+            for inst in instances:
+                path = win32pdh.MakeCounterPath((None, self._class_name, inst, None, 0, self._instance_name))
+                self.counterdict[inst] = win32pdh.AddCounter(self.hq, path)
```

Comment:
Probably can (and I'll wrap it too, just in case), but shouldn't.  In the single instance case, it can fail because the caller is passing the `instance` name, and so it could be invalid.  In the multi-instance case, we've gotten back the list of valid instances (from EnumInstances), so it *should* never fail.  But it never hurts.